### PR TITLE
Only run coverage with inlining on

### LINKED ISSUE #55: Split noinline test and coverage
Turning off inlining should no longer be needed for coverage so we can remove it for faster coverage run.

If we want to make sure all tests passes with inlining off we can add another parallel buildbot for that.

@staticfloat

### PR DESCRIPTION
**Pull request by yuyichao**
Use a separate buildbot for noinline tests

Fix #55 
I realized that the test list is from `CoverageBase.jl` so this requires https://github.com/JuliaCI/CoverageBase.jl/pull/19 
@staticfloat

### CODE REVIEW

File: master/nightly_noinline.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
+###############################################################################
+# Define everything needed to build nightly Julia for noinline test
+###############################################################################
+
+noinline_nightly_scheduler = Nightly(name="Julia Noinline Build", builderNames=["nightly_noinline-x64"], hour=[3], branch="master", onlyIfChanged=True)
+c['schedulers'].append(noinline_nightly_scheduler)
+
+arch = "x86"
```
Comment (tkelman) [inline master/nightly_noinline.py]: do we want this 32 or 64 bit?

Comment (yuyichao) [inline master/nightly_noinline.py]: I make it run only on 64bit to save some time but deleted the wrong one here...... fixed....

**Comment by tkelman** [comment]
can we make this run not on the ubuntu slave? there are too many nightly jobs on that builder and it gets backed up

**Comment by yuyichao** [comment]
I've just been copying the previous config file I touched, where should I run this then?

**Comment by tkelman** [comment]
centos 7 maybe? that only has the one job right now

**Comment by yuyichao** [comment]
Sure, is this the right way to do it?

**Comment by yuyichao** [comment]
I just realized that doing this will not run the tests with inliing off.

**Comment by yuyichao** [comment]
@tkelman @staticfloat do we actually still want to have a buildbot running noinline tests? I think it won't really be testing sth that we want to keep working (now that we don't need inlining off for coverage). I might be useful as a Fuzz test but probably not more than that.

If we still want to have no-inline tests, https://github.com/JuliaLang/julia/pull/18680 should provide an much easier way.

**Comment by tkelman** [comment]
I think it's worth identifying and tracking fragile tests, yes

**Comment by yuyichao** [comment]
In that case https://github.com/JuliaLang/julia/pull/18681....

**Comment by staticfloat** [comment]
It's trivial to add a second Ubuntu buildbot if that's what we need.  Then tasks assigned to an Ubuntu builder will be load-balanced accordingly.

**Comment by tkelman** [comment]
we may want to do that anyway, but we have the centos worker and it's a bit underutilized

**Comment by yuyichao** [comment]
Rebased and updated the script to use the env added in https://github.com/JuliaLang/julia/pull/18680. This still launches the test process manually so that the compile test on master process also has inlining off.

**Pull request merged**